### PR TITLE
gRPC plugin support trace client generic call

### PR DESCRIPTION
**Pull request by hutaishi**
skywalking grpc plugin support trace client generic call. Fix https://github.com/apache/skywalking/issues/8936

- [x] Add a test case for the new plugin, refer to [the doc](https://github.com/apache/skywalking-java/blob/main/docs/en/setup/service-agent/java-agent/Plugin-test.md)
- [x] Update the documentation to include this new feature.
- [x] If it's UI related, attach the screenshots below.

### topology
image

### trace
image

#### only trace grpc generic call
Determine if the request parameter is a DynamicMessage
```java
allArguments[1] instanceof com.google.protobuf.DynamicMessage
```

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/grpc-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/grpc/v1/define/NettyClientStreamInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import net.bytebuddy.matcher.ElementMatcher;
+
+/**
+ * Intercept constructor to obtain server IP.
+ * grpc Low version(eg:1.6.0): use io.grpc.netty.NettyClientStream.
+ * grpc High version(eg:1.33.1): use io.grpc.netty.shaded.io.grpc.netty.NettyClientStream.
+ */
+public class NettyClientStreamInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
+
+    private static final String ENHANCE_CLASS = "io.grpc.netty.NettyClientStream";
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/grpc-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/grpc/v1/define/NettyClientStreamInstrumentation.java:39]: Intercepting netty class in the grpc plugin seems not a good idea. User or other RPC framework could use `NettyClientStream` too. Also, in `NettyClientStreamInterceptor`, all conditions are netty native relative.

This is the major concern for this plugin.

File: apm-sniffer/apm-sdk-plugin/grpc-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/grpc/v1/client/ClientCallImplGenericCallInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    @Override
+    public void onConstruct(EnhancedInstance objInst, Object[] allArguments) throws Throwable {
+        MethodDescriptor methodDescriptor = (MethodDescriptor) allArguments[0];
+        objInst.setSkyWalkingDynamicField(methodDescriptor);
+    }
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+            MethodInterceptResult result) throws Throwable {
+        String asyncCallMethod = (String) ContextManager.getRuntimeContext().get(GENERIC_CALL_METHOD);
```
[the remark this answers is not in the document]
Comment (hutaishi) [reply apm-sniffer/apm-sdk-plugin/grpc-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/grpc/v1/client/ClientCallImplGenericCallInterceptor.java:69]: I will follow this good practice to modify it.

File: test/plugin/scenarios/grpc-generic-call-scenario/config/expectedData.yaml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            skipAnalysis: false
+            refs:
+              - { parentEndpoint: org.apache.skywalking.apm.testcase.grpc.proto.TestServiceDynamic.unary,
+                  networkAddress: '', refType: CrossThread,
+                  parentSpanId: 0, parentTraceSegmentId: not null,
+                  parentServiceInstance: not null, parentService: grpc-provider,
+                  traceId: not null }
+      - segmentId: not null
+        spans:
+          - operationName: org.apache.skywalking.apm.testcase.grpc.proto.TestServiceDynamic.unary
```
Comment (wu-sheng) [inline test/plugin/scenarios/grpc-generic-call-scenario/config/expectedData.yaml:54]: Could you list the operation name format rules? I prefer we could keep consistent with gRPC service name(package + service name). Java package name seems a little strange as prefix.

Comment (hutaishi) [reply test/plugin/scenarios/grpc-generic-call-scenario/config/expectedData.yaml:54]: I reused method of `OperationNameFormatUtil#formatOperationName`.
consistent: package + service name + method.
My integration test case package name is the same as the Java package name.
Java package names are not actually used.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**